src/guake/guake_app.py

```diff
@@ -693,21 +696,27 @@ def set_final_window_rect(self):
         window_rect = screen.get_monitor_geometry(monitor)
 
         if os.environ.get('DESKTOP_SESSION') == "ubuntu":
-            unity_hide = self.client.get_int(KEY('/apps/compiz-1/plugins/'
-                                                 'unityshell/screen0/options/launcher_hide_mode'))
-            # launcher_hide_mode = 1 => autohide
-            if unity_hide != 1:
-                # Size of the icons for Unity in Ubuntu <= 12.04
-                # TODO Ubuntu 12.10 use dconf :
-                # /org/compiz/profiles/unity/plugins/unityshell/icon-size
+
+            # For Ubuntu 12.10 and above, use dconf :
+            # For Ubuntu 12.04 and below, use
+            # see if unity dock is hide => unity_hide
+            # and the width of unity dock. => unity_dock
+            if float(platform.dist()[1])>=12.10:
+                unity_hide = int(subprocess.check_output(
+                    ['/usr/bin/dconf', 'read',
+                     '/org/compiz/profiles/unity/plugins/unityshell/launcher-hide-mode']))
```

Comment:
that's a chicken and egg problem. This should be handled by the packager (for debian, for rehat, ...). But, as a User, I want to be able to use Guake directly from source. And as a Maintainer, I don't want packager to introduce regression in my software.... So I don't see how to handle distribution specific stuff differently.

Maybe a good thing to do is to move this trick in a spefic module so distribution hack are better identified